zenodo/modules/auditor/oai.py

```diff
@@ -71,16 +69,18 @@ def _build_oai_sets_from_db(self):
                 PersistentIdentifier,
                 RecordMetadata.id == PersistentIdentifier.object_uuid)
             .filter(
-                PersistentIdentifier.pid_type == 'recid',
+                PersistentIdentifier.pid_type == 'oai',
                 PersistentIdentifier.status == PIDStatus.REGISTERED)
             .with_entities(
-                cast(PersistentIdentifier.pid_value, sa.Integer),
+                PersistentIdentifier.pid_value,
                 RecordMetadata.json)
         )
 
-        for recid, record in record_oai_sets:
+        for oai_id, record in record_oai_sets:
             for s in record.get('_oai', {}).get('sets', []):
-                self.cache.sadd('{}:{}'.format(self.cache_prefix, s), recid)
+                oai_control_number = oai_id.split(':')[-1]
```

Comment:
One caution on this one that just came in mind: We have OAI IDs like `zenodo.org:123`, but also like `cern.ch:123`. Since we only keep the last part of the identifier (the integer part) if eg. an OAI set contains both `zenodo.org:123` and `cern.ch:123`, our calculated set will only contain one element (`set([1])`), so we might end up with wrong results.